packages/dataviews/src/components/dataviews-filters/filter.tsx

```diff
@@ -422,28 +124,29 @@ function OperatorSelector( {
 											if (
 												_filter.field === filter.field
 											) {
-												// Reset the value only when switching between operators that have different value types.
-												const OPERATORS_SHOULD_RESET_VALUE =
-													[
-														OPERATOR_BETWEEN,
-														OPERATOR_IN_THE_PAST,
-														OPERATOR_OVER,
-													];
-												const shouldResetValue =
-													currentOperator &&
-													( OPERATORS_SHOULD_RESET_VALUE.includes(
+												const currentOpSelectionModel =
```

Comment:
Instead of hard-coding the operator names here, I've made it filter agnostic. It works as before, but it now depends on how the filter works with the data (single value selection, multi value selection, or custom value).

For example, number can switch between the regular single value (number) to a custom format (between operator):

https://github.com/user-attachments/assets/686026ee-f650-4971-b4a0-fd2eb783c1e4

And dates can switch between the regular single value (date) to a custom format (in the past, over):

https://github.com/user-attachments/assets/5ee0326c-2a5d-4c8c-bdb3-26764ebf19da